ci/conda_env_cpp.txt

```diff
@@ -16,6 +16,11 @@
 # under the License.
 
 aws-sdk-cpp=1.11.68
+azure-core-cpp==1.10.3
+azure-identity-cpp==1.6.0
+azure-storage-blobs-cpp==12.10.0
+azure-storage-common-cpp==12.5.0
+azure-storage-files-datalake-cpp==12.9.0
```

Comment:
I used the same versions we get with VCPKG="a42af01b72c28a8e1d7b48107b33e4f286a55ef6"    # 2023.11.20 Release

Reply:
Do you know if such exact pins are needed here? (could also define just a lower bound)

Reply:
They don't need to be exact like this but there are some versions we want to avoid. I mostly did exact pins so its the same as for the vcpkg based builds

Reply:
I'm happy to go with these as lower bounds though if you prefer

Reply:
Changed to lower bounds